build/scripts/test-build-correctness.ps1

```diff
@@ -36,7 +36,31 @@ try {
     # Verify the state of our generated syntax files
     Write-Host "Checking generated compiler files"
     Exec-Block { & (Join-Path $PSScriptRoot "generate-compiler-code.ps1") -test }
+    Write-Host ""
+    
+    # Verfiy the state of creating run settings for optprof
+    Write-Host "Checking run generation for optprof"
+
+    # set environment variables
+    if (-not (Test-Path env:SYSTEM_TEAMPROJECT)) { $env:SYSTEM_TEAMPROJECT = "DevDiv" }
+    if (-not (Test-Path env:BUILD_REPOSITORY_NAME)) { $env:BUILD_REPOSITORY_NAME = "dotnet/roslyn" }
+    if (-not (Test-Path env:BUILD_SOURCEBRANCHNAME)) { $env:BUILD_SOURCEBRANCHNAME = "test" }
+    if (-not (Test-Path env:BUILD_BUILDID)) { $env:BUILD_BUILDID = "42.42.42.42" }
+    if (-not (Test-Path env:BUILD_SOURCESDIRECTORY)) { $env:BUILD_SOURCESDIRECTORY = $RepoRoot }
+    if (-not (Test-Path env:BUILD_STAGINGDIRECTORY)) { $env:BUILD_STAGINGDIRECTORY = $configDir }
+
+    # create a fake BootstrapperInfo.json file
+    $bootstrapperInfoFolder = Join-Path $configDir "MicroBuild\Output"
+    if(!(Test-Path -Path $bootstrapperInfoFolder )){
+        New-Item -ItemType directory -Path $bootstrapperInfoFolder
+    }
+    $bootstrapperInfoPath = Join-Path $bootstrapperInfoFolder "BootstrapperInfo.json"
+    $bootstrapperInfoContent = "[{""VSBuildVersion"":  ""16.0.28309.54""}]"
```

Comment:
Will hard coding the version here get us into trouble down the road?

Reply:
I could use 42.42.42.42 here instead to better indicate that this is a non-real number.  In the signed build this number comes from Micobuild so if this breaks its not anyones fault who checked into this repo.

Reply:
This number is just read and copied verbatim into the runsettings file